src/tests/ftest/server/daos_server_config.yaml

```diff
@@ -0,0 +1,192 @@
+# change host names to your reserved nodes, the
+# required quantity is indicated by the placeholders
+hosts:
+  test_servers:
+    - server-A
+timeout: 300
+
+server_config:
+  name: daos_server
+
+server_config_val: !mux
+  name_special_chars:
+    config_val:
+      - "name"
+      - "! @#$%^&*()_+{}|:<>?-=[];',./"
+      - 1
+      - "gen"
+  name_alphanumeric:
+    config_val:
+      - "name"
+      - "ABCD1234abcd"
+      - 0
+      - "gen"
+  name_numeric:
+    config_val:
+      - "name"
+      - "12345678"
+      - 0
+      - "gen"
+  name_combo:
+    config_val:
+      - "name"
+      - "AB-C&__dal1234"
+      - 0
+      - "gen"
+  name_list:
+    config_val:
+      - "name"
+      - "amjustin, 12345, abcdef"
+      - 1
+      - "gen"
+  access_point_repeated:
+    config_val:
+      - "access_points"
+      - "[localhost:10001, localhost:10001]"
+      - 1
+      - "gen"
+  access_point_wrong_port:
+    config_val:
+      - "access_points"
+      - "[localhost:12345]"
+      - 1
+      - "gen"
+  access_point_wrong_name:
+    config_val:
+      - "access_points"
+      - "[wrongName:10001]"
+      - 1
+      - "gen"
+  access_point_alphanumeric:
+    config_val:
+      - "access_points"
+      - "[AB12cd:10001]"
+      - 1
+      - "gen"
+  control_log_file_does_not_exist:
+    config_val:
+      - "control_log_file"
+      - "/tmp/does_not_exist.log"
+      - 1
+      - "gen"
+  control_log_file_special_str:
+    config_val:
+      - "control_log_file"
+      - "ABC12cd*()_-"
+      - 1
+      - "gen"
+  control_log_mask_wrong_mask:
+    config_val:
+      - "control_log_mask"
+      - "NOTDEBUG"
+      - 1
+      - "gen"
+  control_log_mask_list:
+    config_val:
+      - "control_log_mask"
+      - "INFO,DEBUG,ERR"
+      - 1
+      - "gen"
+  control_log_mask_special_str:
+    config_val:
+      - "nr_hugepages"
+      - "ABC12cd*()_-"
+      - 1
+      - "gen"
+  nr_hugepages_str:
+    config_val:
+      - "nr_hugepages"
+      - "ABCdef"
+      - 1
+      - "gen"
+  # nr_hugepages_overflow_int:
```

Comment:
These commented cases are cases that might be behaving incorrectly, i.e. the server starts up when it shouldn't. For example, here the nr_hugepages config value is set to overflow the int32 variable that is setup in the go code, the server starts up and ignores the huge value. Should this be handled as a bug or should cases like these should be added at all? @mjmac, @tanabarr any feedback on such cases?

Reply:
I suspect that this is an artifact of the yaml library we're using to parse the config files. Probably worth looking at to see if we can make it give us an error that we display to the user rather than silently ignoring bogus values. Assuming that there isn't anything we can change, though, I'm not that there's anything for us to do. If the value is set to zero or unspecified, then the defaults are used. This seems like reasonable behavior to me.

Reply:
Sounds good. Thanks for the clarification!